src/octoprint/printer/standard.py

```diff
@@ -669,8 +679,19 @@ def add_sd_file(self, filename, absolutePath, on_success=None, on_failure=None,
 		else:
 			# probably something else added through a plugin, use it's basename as-is
 			remoteName = os.path.basename(filename)
-		self._create_estimator("stream")
-		self._comm.startFileTransfer(absolutePath, filename, "/" + remoteName,
+			
+		if len(self.sd_card_upload_hooks.items()) > 0:
```

Comment:
Hooks that replace existing functionality so far always have a return value that indicates whether the hook took care of things or the default should be applied. See for example the `serial_factory` hook. This should be done here as well so that we really always run the default if no hook volunteered.

Reply:
mmm OK, as I understand it, I have to add a default method to the existing hook list, like that?
serial_factories = self._serial_factory_hooks.items() + [("default", default)]

Reply:
it's change :)